django/db/models/lookups.py

```diff
@@ -143,6 +144,18 @@ def contains_over_clause(self):
     def is_summary(self):
         return self.lhs.is_summary or getattr(self.rhs, 'is_summary', False)
 
+    @property
+    def identity(self):
+        return self.lhs, self.rhs
+
+    def __eq__(self, other):
+        if not isinstance(other, self.__class__):
+            return NotImplemented
+        return self.identity == other.identity
+
+    def __hash__(self):
+        return hash(make_hashable(self.identity))
+
```

Comment:
@ngnpope `Lookup` are a good example of why including `self.__class__` in identity is so important.

If it's not then `Exact(Col('col'), Value(False)) == In(Col('col'), Value(False))`